lib/sidekiq_unique_jobs/lock/until_and_while_executing_reschedule.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module SidekiqUniqueJobs
+  class Lock
+    # Locks jobs while the job is executing in the server process
+    # - Locks on perform_in or perform_async (see {UntilExecuting})
+    # - Unlocks before yielding to the worker's perform method (see {UntilExecuting})
+    # - Locks before yielding to the worker's perform method (see {WhileExecuting})
+    # - Unlocks after yielding to the worker's perform method (see {WhileExecuting})
+    #
+    # See {#lock} for more information about the client.
+    # See {#execute} for more information about the server
+    #
+    # @author Maciej Mucha <maciej@northpass.com>
+    class UntilAndWhileExecutingReschedule < BaseLock
+      # Executes in the Sidekiq server process
+      # @yield to the worker class perform method
+      def execute
+        return unless locked?
```

Comment:
@mfly I assume this is here because it's in `UntilAndWhileExecuting`. @marclennox could you provide any additional context for why this line is needed? I looked at https://github.com/mhenrixon/sidekiq-unique-jobs/pull/277 which added it but don't really understand.

I'm asking because this line appears to break [Sidekiq Pro's `super_fetch` reliability feature](https://github.com/mperham/sidekiq/wiki/Reliability#using-super_fetch). When I start a job using this lock strategy and then kill the worker suddenly with `kill -9`, `super_fetch` will see that the job died and try to re-enqueue it but won't be able to because of this line. When I remove this `return unless locked?` line it works as I expect.

Reply:
To be honest I haven't worked with this library in many years and have lost any context in the deep recesses of my brain...  apologies.

Reply:
No worries! I actually think I meant to tag @mhenrixon for that question :)